lib/signaling/v2/cancelableroomsignalingpromise.js

```diff
@@ -12,120 +12,133 @@ const {
 
 const { flatMap } = require('../../util');
 
-function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, iceServerSource, encodingParameters, preferredCodecs, options) {
+function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, encodingParameters, preferredCodecs, options) {
   options = Object.assign({
     PeerConnectionManager: DefaultPeerConnectionManager,
     RoomV2: DefaultRoomV2,
     Transport: DefaultTransport
   }, options);
 
-  let transport;
-
-  const PeerConnectionManager = options.PeerConnectionManager;
-  const RoomV2 = options.RoomV2;
-
-  const peerConnectionManager = new PeerConnectionManager(iceServerSource, encodingParameters, preferredCodecs, options);
-
+  const { PeerConnectionManager, RoomV2, Transport, iceServers } = options;
+  const peerConnectionManager = new PeerConnectionManager(encodingParameters, preferredCodecs, options);
   const trackSenders = flatMap(localParticipant.tracks, trackV2 => [trackV2.trackTransceiver]);
-
-  peerConnectionManager.setConfiguration(options);
   peerConnectionManager.setTrackSenders(trackSenders);
 
   const cancelationError = new Error('Canceled');
 
-  return new CancelablePromise(function onCreate(resolve, reject, isCanceled) {
-    peerConnectionManager.createAndOffer().then(function createAndOfferSucceeded() {
-      // NOTE(mmalavalli): PeerConnectionManager#createAndOffer() queues the
-      // initial offer in the event queue for the 'description' event. So,
-      // we are dequeueing to prevent the spurious 'update' message sent by
-      // the client after connecting to a room.
-      peerConnectionManager.dequeue('description');
+  let transport;
 
-      return new Promise((resolve, reject) => {
+  return new CancelablePromise((resolve, reject, isCanceled) => {
+    return new Promise((resolve, reject) => {
```

Comment:
I remember you making a suggestion of resolving icePromise locally, instead of having transport always send onIced on my PR - which I think made this code much simpler. Why not take the same route? 
```
const ensureIceServers = iceServers ? Promise.resolve(iceServers) : waitForEvent(transport, 'iced');
```

Reply:
@makarandp0 You were emitting an "iced" event on the Transport and then calling a separate function called `resumeConnect()`. In this case, there is no need for `resumeConnect()` since `onIced` returns a Promise which can be used by the Transport to resume connecting.

Reply:
ah makes sense. I think I like this approach too. But I do have some suggestion below to make the handling of cancellable promise and OnIced more streamlined.